### PR TITLE
Fev 417 plugin name

### PR DESCRIPTION
**Pull request by alanovyy**
command to run: create-playkit-plugin

### CODE REVIEW

**Comment by kaltura-hooks** [comment]
Hi @alanovyy,
Thank you for contributing this pull request!
Please sign the Kaltura CLA so we can review and merge your contribution.
Learn more at http://bit.ly/KalturaContrib

File: packages/create-playkit-plugin/createPlaykitPlugin.js [hunk 1]
```diff
@@ -129,19 +133,68 @@ if (program.info) {
...
+        name: "npmName",
+        message: "Please specify the plugin NPM name ( default to @playkit-js/{name}-plugin )",
+        validate: validation(/^(@[a-z-]+\/[a-z-]+|[a-z-]+)*$/g),
+        initial: `@playkit-js/${name}-plugin`,
+    })
+
+    const askGithubRepo = name => ({
+        type: "text",
+        name: "githubRepo",
+        message: `Please specify the plugin GitHub repository ( default to kaltura/playkit-js-{name} )`,
```
Comment (esakal) [inline packages/create-playkit-plugin/createPlaykitPlugin.js]: no need to write the `default to...` part, it is the initial value. 

same goes for other questions here

File: packages/create-playkit-plugin/createPlaykitPlugin.js [hunk 2]
```diff
@@ -152,43 +205,57 @@ function printValidationResults(results) {
...
 
-const hiddenProgram = new commander.Command()
-    .option(
-        '--internal-testing-template <path-to-template>',
-        '(internal usage only, DO NOT RELY ON THIS) ' +
-        'use a non-standard application template'
-    )
-    .parse(process.argv);
+(async () => {
+    if (typeof projectName === 'undefined') {
```
Comment (esakal) [inline packages/create-playkit-plugin/createPlaykitPlugin.js]: the questions should be asked regardless if someone provided `projectName`. 

I will modify the jira ticket to better implement this part.
prior for running the prompts, check if user provided `projectName`. if it does, check if this is a path or only a name and use those values as initials for `projectName` and `projectDestination`.

- given `~/dev/temp/abcd` - `projectName=abcd, projectDestination=~/dev/temp/abcd`
- given `abcd` - `projectName=acbd, projectDestination=${process.cwd()}/playkit-js-abcd`

File: packages/create-playkit-plugin/createPlaykitPlugin.js [hunk 3]
```diff
@@ -129,19 +133,74 @@ if (program.info) {
...
-    console.log();
-    console.log('For example:');
-    console.log(`  ${chalk.cyan(program.name())} ${chalk.green('my-react-app')}`);
-    console.log();
-    console.log(
-        `Run ${chalk.cyan(`${program.name()} --help`)} to see all options.`
-    );
-    process.exit(1);
+async function askProjectOptions() {
+    const validation = (pattern = /^[a-zA-Z0-9-]*$/g) => input => {
```
Comment (esakal) [inline packages/create-playkit-plugin/createPlaykitPlugin.js]: I don't think numbers should be accepted here, it this taken from the ticket?

File: packages/create-playkit-plugin/createPlaykitPlugin.js [hunk 4]
```diff
@@ -129,19 +133,74 @@ if (program.info) {
...
-    );
-    console.log();
-    console.log('For example:');
-    console.log(`  ${chalk.cyan(program.name())} ${chalk.green('my-react-app')}`);
-    console.log();
-    console.log(
-        `Run ${chalk.cyan(`${program.name()} --help`)} to see all options.`
-    );
-    process.exit(1);
+async function askProjectOptions() {
```
Comment (esakal) [inline packages/create-playkit-plugin/createPlaykitPlugin.js]: general comment
pressing `ctrl+c` to exit the process while asking the questions just chose the default value instead, it doesn't respect the signal to exit the program

image

File: packages/create-playkit-plugin/createPlaykitPlugin.js [hunk 5]
```diff
@@ -129,19 +133,74 @@ if (program.info) {
...
+        message: `Please specify the project name:`,
+        validate: validation(),
+        initial: projectName,
+    };
+
+    const askNpmName = name => ({
+        type: "text",
+        name: "npmName",
+        message: "Please specify the plugin NPM name:",
+        validate: validation(/^(@[a-z-]+\/[a-z-]+|[a-z-]+)*$/g),
```
Comment (esakal) [inline packages/create-playkit-plugin/createPlaykitPlugin.js]: I'm not sure the asterisk is relevant here

image

File: packages/create-playkit-plugin/createPlaykitPlugin.js [hunk 6]
```diff
@@ -129,19 +133,74 @@ if (program.info) {
...
+        initial: `@playkit-js/${name}-plugin`,
+    });
+
+    const askGithubRepo = name => ({
+        type: "text",
+        name: "githubRepo",
+        message: `Please specify the plugin GitHub repository:`,
+        validate: validation(/^([a-z-]+\/[a-z-]+|[a-z-]+)*$/g),
```
Comment (esakal) [inline packages/create-playkit-plugin/createPlaykitPlugin.js]: same here

File: packages/create-playkit-plugin/createPlaykitPlugin.js [hunk 7]
```diff
@@ -129,19 +133,74 @@ if (program.info) {
...
+        initial: `kaltura/playkit-js-${name}`,
+    });
+
+    const askDestination = (name) => ({
+        type: "text",
+        name: "destination",
+        message: `Please specify the destination folder where the plugin will be initialized:`,
+        initial: projectDestination || `${process.cwd()}/playkit-js-${name}`,
```
Comment (esakal) [inline packages/create-playkit-plugin/createPlaykitPlugin.js]: you should validate that the value is structured correctly

**Pull request merged**